packages/react/src/components/DashboardEditor/DashboardEditorCardRenderer.jsx

```diff
@@ -311,42 +189,30 @@ const DashboardEditorCardRenderer = React.memo(
         onDuplicate,
         onRemove,
         isSelected,
-        onShowImageGallery
+        onShowImageGallery,
+        availableDimensions
       );
     }
 
     // otherwise use our default renderers
-    switch (cardProps.type) {
-      case CARD_TYPES.VALUE:
-        return renderValueCard(cardProps);
-      case CARD_TYPES.TIMESERIES:
-        return <EditorTimeSeriesCard {...cardProps} />;
-      case CARD_TYPES.BAR:
-        return (
-          <EditorBarChartCard
-            {...cardProps}
-            dataItems={dataItems}
-            availableDimensions={availableDimensions}
-          />
-        );
-      case CARD_TYPES.TABLE:
-        return renderTableCard(cardProps);
-      case CARD_TYPES.IMAGE:
-        return renderImageCard({
-          ...cardProps,
-          values: {
-            ...cardProps.values,
-            hotspots: cardProps.values?.hotspots?.concat(dynamicHotspots),
-          },
-        });
-      case CARD_TYPES.LIST:
-        return renderListCard(cardProps);
-      default:
-        // if the user passes an element for a custom card type, render it
-        return React.isValidElement(cardProps.content) || typeof cardProps.content === 'function'
-          ? renderCustomCard(cardProps)
-          : renderDefaultCard(cardProps);
-    }
+    // If it's an image card we need to update its dynamic hotspots
+    const card = useMemo(
+      () =>
+        cardProps.type !== CARD_TYPES.IMAGE
+          ? cardProps
+          : {
+              ...cardProps,
+              values: {
+                ...cardProps.values,
+                hotspots: cardProps.values?.hotspots?.concat(dynamicHotspots),
```

Comment:
This seems to be a little different than the old renderImageCard function. Previously, we were concatting the dynamicHotspots before passing to the renderImageCard, but then, the renderImageCard would filter the hotspots. Is this an intentional change?

see: https://github.com/carbon-design-system/carbon-addons-iot-react/pull/2790/files#diff-2ff08df7c609521e34782529133308227cf907b56069ea653c8b6ef6eeafed77L102
```
hotspots: props.values?.hotspots?.filter((hotspot) => hotspot.type !== 'dynamic') || [],
```

Reply:
no, this is an error, thanks for catching @kevinsperrine .  I'll make sure to filter those again

Reply:
here you go

Reply:
I think we might be missing a push. I still don't see the filter.